src/nvim/lua/vim.lua

```diff
@@ -666,4 +666,15 @@ vim._expand_pat_get_parts = function(lua_string)
   return parts, search_index
 end
 
+---@private
+function vim._pretty_print(...)
```

Comment:
Why not make this public?

Reply:
inspect module is already exposed as `vim.inspect` and I wanted to use `vim._pretty_print` as a helper for this . Though I have no issue in exposing this to user as `vim.pretty_print` if that's what's wanted . I that case we should return the args too so it can be placed in middle of expressions.